ml-dhs-aws/src/main/java/com/marklogic/client/dhs/JobRunner.java

```diff
@@ -64,20 +95,42 @@ public void run(DatabaseClient client, InputStream csvRecords, InputStream jobCo
                         throwable.printStackTrace(System.out);
                      });
 
-         ObjectMapper objectMapper = new ObjectMapper();
-         ObjectNode jobControlObj = (ObjectNode) objectMapper.readTree(jobControl);
-         ObjectNode jobDef        = (ObjectNode) jobControlObj.get("job");
-         ObjectNode recordDef     = (ObjectNode) jobControlObj.get("record");
 
          ObjectLoader loader = new ObjectLoader(batcher, recordDef, getJobDirectory(), documentMetadata);
          CSVConverter converter = new CSVConverter();
 
-// TODO: write before job document in /jobStart and /jobs/ID collections or send before job payload to DHF endpoint
 
+
+         Iterator<ObjectNode> itr = converter.convertObject(csvRecords);
+         if(!itr.hasNext()) {
+        	 throw new MarkLogicIOException("No header found.");
+         }
+         ObjectNode csvNode = itr.next();
+         
+         Iterator<String> csvHeader = csvNode.fieldNames();
+         String headerValue = csvNode.get(csvHeader.next()).toString();
```

Comment:
Isn't csvNode the first data row rather than the header row?  If so, we want to add the field names and not the field values.  To put it another way, doesn't line 111 return the value of the first field in the first data row?